rex-engine/src/codec.rs

```diff
@@ -472,6 +505,49 @@ where
     }
 }
 
+impl<T, E> Decode for Result<T, E>
+where
+    T: Decode,
+    E: Decode,
+{
+    fn try_decode(v: &Expr) -> Result<Self, Error> {
+        match v {
+            Expr::Named(_id, _span, name, x) if name == "Ok" => {
+                Ok(Ok(T::try_decode(x)?))
+            }
+            Expr::Named(_id, _span, name, x) if name == "Err" => {
+                Ok(Err(E::try_decode(x)?))
+            }
+            _ => Err(Error::ExpectedTypeGotValue {
+                expected: Type::Result(
+                    Box::new(Type::Var(Id::new())),
+                    Box::new(Type::Var(Id::new()))),
```

Comment:
instead of doing this, you should require that `T: ToType` and `E: ToType` and then you can
```suggestion
                    Box::new(T::to_type()),
                    Box::new(E::to_type()),
```
for a better error message